app_dart/test/request_handlers/get_benchmarks_test.dart

```diff
@@ -39,106 +39,210 @@ void main() {
+          timestamp: 1,
+          branch: 'master');
+      config.db.values[timeSeriesValue1.key] = timeSeriesValue1;
+      config.db.values[timeSeries.key] = timeSeries;
+      config.db.values[commit1.key] = commit1;
+      handler = GetBenchmarks(
+        config,
+        datastoreProvider: (DatastoreDB db) => DatastoreService(config.db, 5),
       );
+
+      final Map<String, dynamic> result = await decodeHandlerBody();
+      final List<dynamic> benchmarks = result['Benchmarks'] as List<dynamic>;
+      final Map<String, dynamic> benchmark =
+          benchmarks.first as Map<String, dynamic>;
+
+      expect(benchmark['Values'].length, 1);
+    });
+
+    test('returns only maxRecordsValue commits even though there are more',
+        () async {
+      config.maxRecordsValue = 2;
+      final TimeSeries timeSeries = TimeSeries(
+          key: config.db.emptyKey.append(TimeSeries, id: 'test.test1'));
+
+      final TimeSeriesValue timeSeriesValue1 = TimeSeriesValue(
+          key: timeSeries.key.append(TimeSeriesValue, id: 1),
+          value: 1,
+          branch: 'master');
       final Commit commit1 = Commit(
-          key: config.db.emptyKey.append(Commit,
-              id: 'flutter/flutter/ea28a9c34dc701de891eaf74503ca4717019f829'),
-          timestamp: 3,
+          key: config.db.emptyKey.append(Commit, id: 'abc'),
+          timestamp: 1,
           branch: 'master');
       final Commit commit2 = Commit(
-          key: config.db.emptyKey.append(Commit,
-              id: 'flutter/flutter/d5b0b3c8d1c5fd89302089077ccabbcfaae045e4'),
-          timestamp: 1,
+          key: config.db.emptyKey.append(Commit, id: 'def'),
+          timestamp: 2,
+          branch: 'master');
+      final Commit commit3 = Commit(
+          key: config.db.emptyKey.append(Commit, id: 'ghi'),
+          timestamp: 3,
           branch: 'master');
-      config.db.values[timeSeriesValue.key] = timeSeriesValue;
+      config.db.values[timeSeriesValue1.key] = timeSeriesValue1;
       config.db.values[timeSeries.key] = timeSeries;
       config.db.values[commit1.key] = commit1;
       config.db.values[commit2.key] = commit2;
+      config.db.values[commit3.key] = commit3;
       handler = GetBenchmarks(
         config,
         datastoreProvider: (DatastoreDB db) => DatastoreService(config.db, 5),
       );
 
-      expect(config.db.values.length, 4);
-
       final Map<String, dynamic> result = await decodeHandlerBody();
-
-      expect(result['Benchmarks'].length, 1);
       final List<dynamic> benchmarks = result['Benchmarks'] as List<dynamic>;
       final Map<String, dynamic> benchmark =
           benchmarks.first as Map<String, dynamic>;
+
       expect(benchmark['Values'].length, 2);
     });
 
-    test('reports statuses with input branch', () async {
+    test(
+        'returns only release branch commits when there are more release branch commits than the maxRecordsValue - with input branch',
```

Comment:
Here and below, let's try to keep test names short. If we need to add full descriptions let's add it as docs.

Reply:
Use shorter test names, and put extra words to doc.